### PR TITLE
Reusable GHA for Terraform deploy

### PR DESCRIPTION
**Pull request by derekadombek**
## Description
This is so we can finally automate the Terraform apply process with Github Actions rather than from my personal command line. This will be added to the combined deployment.

## Related Issues
[Link any related issues or tasks from your project management system.]

## Checklist
- [x] The title of this PR is descriptive and concise.
- [x] My changes follow the style guidelines of this project.
- [x] I have added or updated test cases to cover my changes.
- [x] I've let the team know about this PR by linking it in the review channel

### CODE REVIEW

File: ops/terraform/config/dev.config [hunk 1]
```diff
@@ -1,4 +1,4 @@
-    storage_account_name = "tfstaterv2024"
-    container_name       = "rv-tfstate"
-    key                  = "dev.terraform.tfstate"
-    use_oidc             = true
```
Comment (derekadombek) [inline ops/terraform/config/dev.config:4]: these are just auto formating changes

File: ops/terraform/config/dev.config [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
\ No newline at end of file
+storage_account_name = "tfstaterv2024"
```
Comment (arinkulshi-skylight) [inline ops/terraform/config/dev.config:1]: Assuming these are for each of the different dev environments?

**Review by arinkulshi-skylight — APPROVED** [review]
LGTM! Would love to test out the e2e flow when it is ready.

File: ops/terraform/config/dev.config (hunk 2 above)
Comment (derekadombek) [reply ops/terraform/config/dev.config:1]: that correct! so with TF you can store the configuration state files either locally on our computers or a in a central location. these config files basically create and update these files from a global AZ Resource Group in the same account subscription. the benefit of seperating the out into individual files like this is so we can use the `-backend-config=config/${{ inputs.deploy-env }}.config` flag in the `terraform` cli. We are no using that here in a reusable GHA: https://github.com/CDCgov/ReportVision/blob/fcdfeeb5dd0026075aa3e589432f0ccc9bbe6c14/.github/actions/tf-setup/action.yml#L53

**Comment by derekadombek** [comment]
> LGTM! Would love to test out the e2e flow when it is ready.

💯 ! Getting very close for sure.

**Pull request merged**